python/cuda_parallel/cuda/parallel/experimental/_iterators.py

```diff
@@ -96,18 +96,18 @@ def __init__(self, ptr, ntype):
         data_as_ntype_pp = numba.types.CPointer(numba.types.CPointer(ntype))
         self.ntype = ntype
         self.prefix = "pointer_" + ntype.name
-        self.ltoirs = [
+        self.ltoirs, _ = zip(*[
             _ncc(
                 f"{self.prefix}_advance",
                 RawPointer.pointer_advance,
-                numba.types.void(data_as_ntype_pp, _DISTANCE_NUMBA_TYPE),
```

Comment:
In general, the return type can (and probably should) be left to the compiler to infer.